pkg/nad/masterplugin.go

```diff
@@ -393,3 +393,161 @@ func (plugin *MasterIPVlanPlugin) GetMasterPluginConfig() (*MasterPlugin, error)
+	validModes := map[string]bool{
+		"balance-rr":    true,
+		"active-backup": true,
+		"balance-xor":   true,
+	}
+
+	if mode == "" {
+		glog.V(100).Infof("error adding empty mode to MasterBondPlugin")
+
+		return nil
+	}
+
+	// Check if the provided mode is valid
+	if !validModes[mode] {
+		glog.V(100).Infof("error: invalid mode type used for MasterBondPlugin bond interface")
+
+		return nil
+	}
+
+	plugin.masterPlugin.Mode = mode
+
+	return plugin
+}
+
+// WithLinksInContainer defines nad with linksInContainer as true.
+func (plugin *MasterBondPlugin) WithLinksInContainer(linksInContainer bool) *MasterBondPlugin {
+	if !linksInContainer {
+		glog.V(100).Infof("error adding empty linksInContainer to MasterBondPlugin")
+
+		plugin.errorMsg = "error adding empty linksInContainer to MasterBondPlugin"
+	}
+
+	plugin.masterPlugin.LinksInContainer = linksInContainer
+
+	return plugin
+}
+
+// WithFailOverMac defines nad with WithFailOverMac amount.
+func (plugin *MasterBondPlugin) WithFailOverMac(failOverMac int) *MasterBondPlugin {
+	if failOverMac == 0 {
+		glog.V(100).Infof("error adding empty FailOverMac to MasterBondPlugin")
+
+		plugin.errorMsg = "error adding empty FailOverMac to MasterBondPlugin"
+	}
+
+	plugin.masterPlugin.FailOverMac = failOverMac
+
+	return plugin
+}
+
+// WithMiimon defines nad with WithMiimon.
+func (plugin *MasterBondPlugin) WithMiimon(miimon string) *MasterBondPlugin {
+	if miimon == "" {
+		glog.V(100).Infof("error adding empty miimon to MasterBondPlugin")
+
+		plugin.errorMsg = "error adding empty miimon to MasterBondPlugin"
+	}
+
+	plugin.masterPlugin.Miimon = miimon
+
+	return plugin
+}
+
+// WithLinks defines nad with configuration for Links.
+func (plugin *MasterBondPlugin) WithLinks(links []Link) *MasterBondPlugin {
+	if links == nil {
+		glog.V(100).Infof("error adding empty links to MasterBondPlugin")
+
+		plugin.errorMsg = "error adding empty links to MasterBondPlugin"
+	}
+
+	plugin.masterPlugin.Links = links
+
+	return plugin
+}
+
+// WithCapabilities defines nad with configuration for capabilities.
+func (plugin *MasterBondPlugin) WithCapabilities(capabilities *Capability) *MasterBondPlugin {
+	if capabilities == nil {
+		glog.V(100).Infof("error adding empty capabilities to MasterBondPlugin")
+
+		plugin.errorMsg = "error adding empty capabilities to MasterBondPlugin"
+	}
+
+	plugin.masterPlugin.Capabilities = capabilities
+
+	return plugin
+}
+
+// WithMTU defines nad with MTU.
+func (plugin *MasterBondPlugin) WithMTU(mtu int) *MasterBondPlugin {
+	if 1 > mtu || mtu > 9192 {
+		glog.V(100).Infof("error adding incorrect mtu value to MasterBondPlugin")
+
+		plugin.errorMsg = fmt.Sprintf("invalid mtu size %d allowed mtu should be in range 1...9192", mtu)
+	}
+
+	plugin.masterPlugin.Mtu = mtu
+
+	if plugin.errorMsg == "" {
+		plugin.errorMsg = "An error occurred applying the mtu value"
+	}
+
+	return plugin
+}
+
+// WithIPAM defines nad with IP type.
```

Comment:
// WithIPAM defines IPAM configuration to MasterBondPlugin

Reply:
changed